x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/field_parameters/reference_field_selects.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import React, { useEffect, useMemo } from 'react';
+
+import { useLocation } from 'react-router-dom';
+import {
+  FieldHook,
+  Form,
+  FormDataProvider,
+  SuperSelectField,
+  UseField,
+  useForm,
+} from '../../../shared_imports';
+import { SuperSelectOption } from '../../../types';
+import { useLoadIndexMappings } from '../../../../../services';
+import { getFieldConfig } from '../../../lib';
+
+interface Props {
+  onChange(value: unknown): void;
+  'data-test-subj'?: string;
+}
+
+export const ReferenceFieldSelects = ({ onChange, 'data-test-subj': dataTestSubj }: Props) => {
+  const { form } = useForm({ defaultValue: { main: 'body-content' } });
+  const { subscribe } = form;
+
+  const { search } = useLocation();
+  const queryParams = useMemo(() => new URLSearchParams(search), [search]);
+  const { data } = useLoadIndexMappings(queryParams.get('indexName') ?? '');
```

Comment:
I think this api call could be avoided if fields of text type  are filtered from [previous state fields ](https://github.com/elastic/kibana/blob/05d5d47a6722a858c08acfd9f332395a6968ec58/x-pack/plugins/index_management/public/application/sections/home/index_list/details_page/details_page_mappings_content.tsx#L102) and is passed as a props to child components.  Not a blocker, probably an alternate approach to avoid this api call